### DISCUSSION TITLE
用 Vim 编辑 Markdown 时直接粘贴图片 — 码志

### DISCUSSION THREAD

**Discussion by mzlogin** [mzlogin/blog-comments#221] [Announcements]
https://mazhuang.org/2020/05/04/paste-image-in-vim/

 我习惯使用 Vim 编辑 Markdown 文件，一直存在一个痛点就是粘贴图片很不方便。

**Comment by Augists** [comment]
## 图片存储

截图的默认存储路径会不会太乱了，而且怎么截完图就知道图片名字的呢？不是很明白

## 我的方法

截图进剪贴板，iPic从剪贴板上传图床，vim里`inoremap` 给markdown文件插入`![]()`然后直接粘贴

**Comment by mzlogin** [comment]
> @Augists
> ## 图片存储
> 
> 截图的默认存储路径会不会太乱了，而且怎么截完图就知道图片名字的呢？不是很明白
> 
> ## 我的方法
> 
> 截图进剪贴板，iPic从剪贴板上传图床，vim里`inoremap` 给markdown文件插入`![]()`然后直接粘贴

使用习惯不一样吧，我写博客的一般习惯不使用图床，直接和文章一起放 GitHub 上，图片一般是手动根据图片内容命名，所以知道名称。

**Comment by cheng3100** [comment]
hi，我发现本地使用正常，但如果通过ssh登录到服务器就粘贴不上了，会生成空的img目录，这个你有遇到同样现象吗？

**Comment by mzlogin** [comment]
> hi，我发现本地使用正常，但如果通过ssh登录到服务器就粘贴不上了，会生成空的img目录，这个你有遇到同样现象吗？

SSH 应该是用不了的……常见的 SSH 客户端也支持不了从剪贴板粘贴图片吧～这种场景得想想其它办法了。

**Comment by Augists** [comment]
> @mzlogin
> > hi，我发现本地使用正常，但如果通过ssh登录到服务器就粘贴不上了，会生成空的img目录，这个你有遇到同样现象吗？
> 
> SSH 应该是用不了的……常见的 SSH 客户端也支持不了从剪贴板粘贴图片吧～这种场景得想想其它办法了。
我用 mac，从系统剪贴板粘贴上图床的地址没啥问题啊
不知道你们说的是不是这个

**Comment by mzlogin** [comment]
> > @mzlogin
> > > hi，我发现本地使用正常，但如果通过ssh登录到服务器就粘贴不上了，会生成空的img目录，这个你有遇到同样现象吗？
> > 
> > 
> > SSH 应该是用不了的……常见的 SSH 客户端也支持不了从剪贴板粘贴图片吧～这种场景得想想其它办法了。
> > 我用 mac，从系统剪贴板粘贴上图床的地址没啥问题啊
> > 不知道你们说的是不是这个

应该不是一回事，图床地址是个 URL（文本），直接贴图片那是要向服务器传文件/二进制流了。
